astropy/io/fits/tests/test_image.py

```diff
@@ -73,7 +73,7 @@ def test_constructor_copies_header(self):
         # Original header should be unchanged
         assert phdr['FILENAME'] == 'labq01i3q_rawtag.fits'
 
-    @raises(ValueError)
+    @raises(IndexError)
```

Comment:
@embray - `self._read_all` is set to True when the file is closed (https://github.com/astropy/astropy/blob/master/astropy/io/fits/hdu/hdulist.py#L1146). ~~So it would be necessary to keep the part of the code deleted in this PR which computes `self._total_read` ?~~ 
There must be a way to know if the file was fully read (with the `EOFError` block I guess), but currently it is not possible, so the commit I just pushed modifies the exception message unconditionally for now. 

Maybe it's good enough ? I mean is it useful to add more complexity to something already quite complex, just to get a more correct exception message for a very specific case.